### ISSUE TITLE
[1.16.5]Not issue but questions

### ISSUE THREAD

**Issue by BLSK** [McJty/InControl#236]
A old mod I use to use had a fair bit yours has but dabbling and coming across a few things on the wiki as I am testing and have questions that maybe can be answered as this mod seems far more powerful than the 1.7.10 counterpart I am thinking of.

Expressions
In some of the conditions it is possible to use expressions. An expression is basically a string specifying some test to do on an integer value. Here are a few examples:
>10: evaluate to true if the number we are testing on is greater than 10
!=10: evaluate to true if it is different from 10
4-50: evaluate to true if the number is between 4 and 50 (inclusive)
10: evaluate to true if the number is equal to 10
The following comparators are supported: >, >=, <, <=, =, and !=.

**"dimension": "minecraft:overworld",
dimension: a single dimension or list of dimensions (like 'minecraft:overworld'). This is required. If you don't specify any dimensions then nothing will **happen**
Numerical dimension ID's no longer work (like 0 for the overworld. Use 'minecraft:overworld' instead). This is also possible in the 1.15.2 versions of InControl and FxControl (and recommended). But in 1.16.3 and later the numerical ID's are gone.**
I read this and wonder... do I use something like DIM2 where there is technically no name and if right would hope wouldn't have to exist prior to it but with the Expressions above would DIM2-DIM10 work or would they all have to be separate or is DIM2 not considered "A Name" as overworld would be considered DIM0(so to speak)
**difficulty: a string with one of the following values: easy, normal, hard, peaceful**
can difficulty take some kind of argument like "playername" overriding difficulty to a difficulty that is different when player is present aka not easy not normal not hard and not peaceful but whole different set of script when player present in the minecraft save or maybe even dimension player is in
not sure but does 'onjoin' have this power as I like to test at a separate difficulty or play at a different difficulty that is much harder than what I specify by default difficulties but at same time I kind of wonder the other things too.

**passive (only for In Control rule files): a boolean that matches passive mobs**
Can passives take some kind of 'augment' like 'when spawned in world % chance to be hostile or neutral

So dabbling in the mod and working on a file to control certain things, but of course these questions came up that I am 100% sure how to deal with as my knowledge is kind of novice on the syntaxes and whatnot of Minecraft and potentially most games but from prior learning and understanding these things from before stood out as possibly powerful things I am trying to explore but not sure how I may go about it as the syntaxes I knew aren't the same and maybe there is some kind of work around to make it possible?

I know it is a lot of wrote... but I am intriqued by 'InControl!' as it presents a possibility for the 1.7.10 Modpack I put together with the other mod a chance to see the same 'Wonder' that that modpack had.

If any conversation to be had just ask and I am fairly sure I can answer(or not... still working on porting a mod of mine to 1.16.5)

**Comment by McJty** [comment]
Dimensions *always* have a proper name. That's required in 1.16 now. So there will never be a DIM2 or anything. Numerical expressions will also no longer work with dimensions as they don't have a numerical ID anymore.

The difficulty is the difficulty that the world is set too. It's not player specific.

What do you mean about passive? A mob is either passive or not. In Control cannot modify if mobs are passive or not. That's up to the mob itself. i.e. a cow is always passive.

**Comment by BLSK** [comment]
so I have to figure out how what the 'actual' name of a dimension is... hmm this may take some digging I guess however after looking at your tutorial you made on a prior version(was looking at your youtube) I saw a few things I didn't notice explained about the spawning that your wiki presented.

The dimension creation device though which is from RFtools... and only just noticed 'you are that mod author' maybe you have the answer to the naming convention... not that I can't dig for it and yes I know it implies 'alpha' too


As for Passives... guess can't add a weight to convert a passive to hostile... shame but thanks for answering.  Guess time to go back to beating my head off the wall while I let your wiki sink in and figure out this name thing if you didn't answer :)

**Comment by McJty** [comment]
RFTools Dimensions are named 'rftoolsdim:whatever' with whatever the name that you picked when building the dimension (I assume you are talking about 1.16.4+)

**Comment by BLSK** [comment]
Yes...(suddenly questioning something)  If I recall you didn't have to name a dimension... it had an auto naming at least back when I last used it which was like.... 1.10...  is the name specifically set by the player or does it's dimension number have any weight on it... (i know the numbers don't work now[still not 100% sure how it has changed so drastically]) hmm god been many years since questions like these have I posed to a mod author... hmm

I guess I may need to do more testing as it has been years and been a while since I used your dimension mod but surprised by the changes of MC I guess more... hmm how to make a good work around...

**Comment by McJty** [comment]
In 1.16 you have to name the dimension

**Comment by BLSK** [comment]
Yes I get that... thinking about how to made a work around... maybe another mod like Hardcore Questing Mode has a work around I can deal with... just have to disable access to say your Dimension mod until those worlds are accessed...(have to do some testing)  but this has sparked a lot and maybe we can come back to this another time :)... need to do some testing

**Comment by McJty** [comment]
All RFTools Dimensions have 'rftoolsdim' in front of their key. So you can recognize them using that

**Comment by BLSK** [comment]
that's cool... Thanks for the help in understanding the few questions I had... now to head out and try some things... Thanks for your time :)

**Comment by BLSK** [comment]
Know maybe irrelevant to thank you now but figured I would.
Took me a while to get things working out fully and had a friend help me with some of the syntaxing issues I was having after the fact
The little bit you did did fix my the hopes for 'In Control'... still requires a work around but... one thing I have been good at since I was 8yo is writing... going to crack my knuckles and get to story telling as my modpack is a form of a story and added the change you did allows me to only stack on top of the 'backstory' I have already written that is over 30 pages long... granted still not finished with the backstory but I am too creative and enjoy spending the webs to anything I touch :)
Thank you McJty... my friend that helped me is a big fan of yours too... webs webs webs... alover
BLSK
